packages/big-design/src/components/PillTabs/PillTabs.tsx

```diff
@@ -1,97 +1,93 @@
 import { CheckIcon, MoreHorizIcon } from '@bigcommerce/big-design-icons';
-import React, { createRef, useCallback, useEffect, useMemo, useState } from 'react';
+import React, { createRef, useMemo } from 'react';
 
-import { useWindowResizeListener } from '../../hooks';
 import { Button } from '../Button';
 import { Dropdown } from '../Dropdown';
 import { Flex } from '../Flex';
 
 import { StyledFlexItem, StyledPillTab } from './styled';
+import { useAvailableWidth } from './useAvailableWidth';
 
 export interface PillTabItem {
   id: string;
   title: string;
 }
 
+interface Pill extends PillTabItem {
+  isVisible: boolean;
+  ref: React.RefObject<HTMLDivElement>;
+}
+
 export interface PillTabsProps {
   items: PillTabItem[];
   activePills: string[];
   onPillClick: (itemId: string) => void;
 }
 
 export const PillTabs: React.FC<PillTabsProps> = ({ activePills, items, onPillClick }) => {
-  const parentRef = createRef<HTMLDivElement>();
-  const dropdownRef = createRef<HTMLDivElement>();
-  const [isMenuVisible, setIsMenuVisible] = useState(false);
-  const [pillsState, setPillsState] = useState(
-    items.map((item) => ({
-      isVisible: true,
-      item,
-      ref: createRef<HTMLDivElement>(),
-    })),
-  );
-
-  const hideOverflowedPills = useCallback(() => {
-    const parentWidth = parentRef.current?.offsetWidth;
-    const dropdownWidth = dropdownRef.current?.offsetWidth;
-
-    if (!parentWidth || !dropdownWidth) {
-      return;
-    }
-
-    let remainingWidth = parentWidth;
-
-    const newState = pillsState.map((stateObj) => {
-      const pillWidth = stateObj.ref.current?.offsetWidth;
-
-      if (!pillWidth) {
-        return stateObj;
-      }
-
-      if (remainingWidth - pillWidth > dropdownWidth) {
-        remainingWidth -= pillWidth;
-
-        return {
-          ...stateObj,
-          isVisible: true,
-        };
-      }
+  const availableWidth = useAvailableWidth();
```

Comment:
🤔 I wonder if this would be better implemented as:


```tsx
const parentRef = useRef();
const dropdownRef = useRef();

const avilableWidth = useAvilableWidth(parentRef, dropdownRef);
```

It looks more straightforward.

Reply:
yep, nicer signature for `useAvailableWidth`

https://github.com/bigcommerce/big-design/pull/1666/commits/02d8d9227aed55ca73c0ad319c0e9e57c16d5c83

Reply:
🤔 should allow you to go broader, hrmldivelement is an htmlelement